Orange/widgets/evaluate/owpredictions.py

```diff
@@ -285,12 +300,16 @@ def _invalidate_predictions(self):
             self.predictors[inputid] = pred._replace(results=None)
 
     def _valid_predictors(self):
-        return [p for p in self.predictors.values()
-                if p.results is not None and not isinstance(p.results, str)]
+        if self.class_var is not None and \
+                self.data is not None:
+            return [p for p in self.predictors.values()
+                    if p.results is not None and not isinstance(p.results, str)]
+        else:
+            return []
 
     def _update_predictions_model(self):
         """Update the prediction view model."""
-        if self.data is not None:
+        if self.data is not None and self.class_var is not None:
```

Comment:
`self.class_var` comes from predictors and can be `not None` even if no data is present. See https://github.com/janezd/orange3/blob/21ba4679bf19efde4498363a56c69bf3798ea0f9/Orange/widgets/evaluate/owpredictions.py#L224.